base/client.jl

```diff
@@ -125,14 +125,26 @@ function repl_cmd(cmd, out)
     nothing
 end
 
-display_error(er) = display_error(er, [])
-function display_error(er, bt)
-    print_with_color(Base.error_color(), STDERR, "ERROR: "; bold = true)
-    with_output_color(Base.error_color(), STDERR) do io
-        showerror(io, er, bt)
+function display_error(io::IO, er, bt)
+    if !isempty(bt)
+        st = stacktrace(bt)
+        if !isempty(st)
+            io = redirect(io, log_error_to, st[1])
+        end
+    end
+    print_with_color(Base.error_color(), io, "ERROR: "; bold = true)
+    # remove REPL-related frames from interactive printing
+    eval_ind = findlast(addr->Base.REPL.ip_matches_func(addr, :eval), bt)
+    if eval_ind != 0
+        bt = bt[1:eval_ind-1]
+    end
+    with_output_color(Base.error_color(), io) do io
```

Comment:
Previously the `REPL.jl` called `Base.showerror`. Now it calls this method which does not do the same thing as the one you deleted from `REPL.jl`. The result is that errors being printed to the REPL will now be different.

It looks like https://github.com/JuliaLang/julia/blob/962583e81df035b35069797691f274f10378d530/base/multi.jl#L1303 is the only location which calls the `client.jl` version of `display_error`. Updating the `client.jl` to match `REPL.jl` would be the best way forward in my opinion.